src/porepy/numerics/ad/operators.py

```diff
@@ -38,6 +40,62 @@ def _get_shape(mat):
         return mat.shape
 
 
+def _get_previous_time_or_iterate(
```

Comment:
An old form of this recursion was hidden inside the scope of some method in `Operator`
Now it is outside, and generalized to handle the abstraction of `IterativeOperator` and `TimeDependentOperator`